ports/stm32/adc.c

```diff
@@ -335,7 +332,11 @@ STATIC void adc_config_channel(ADC_HandleTypeDef *adc_handle, uint32_t channel)
     #if defined(STM32F0)
     sConfig.SamplingTime = ADC_SAMPLETIME_55CYCLES_5;
     #elif defined(STM32F4) || defined(STM32F7)
-    sConfig.SamplingTime = ADC_SAMPLETIME_15CYCLES;
+    if (__HAL_ADC_IS_CHANNEL_INTERNAL(channel)) {
```

Comment:
I assume this is also needed for older MCUs, if not it can be removed.

Reply:
Yes I think it is.  Looking at the `machine_adc.c` implementation, it does use this higher sample time for F4/F7 MCUs.

I tested this change on PYBv1.0 and it works.